### PR TITLE
Approve and reject disruptions

### PR DESCRIPTION
**Pull request by alvak1**
Testing instruction

- Login as staff user and create a disruption or edit a published disruption. Then submit this disruption for approval
- Login as admin or publisher and the disruption should be available to publish or reject

### CODE REVIEW

**Comment by lj-kpmg** [comment]
Couple of bugs found:
* Disruption info not being updated after approving a pending edit
* After editing a disruption consequence it shows up as a duplicate when viewing it on the detail page
* Disruptions should have a status of "Pending approval" on the view all disruptions page

**Comment by alvak1** [comment]
> Couple of bugs found:
> 
> * Disruption info not being updated after approving a pending edit
> * After editing a disruption consequence it shows up as a duplicate when viewing it on the detail page
> * Disruptions should have a status of "Pending approval" on the view all disruptions page

These are now fixed

File: site/pages/api/cancel-changes.test.ts [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
+        getSession: vi.fn(),
+    }));
+
+    beforeEach(() => {
+        getSessionSpy.mockImplementation(() => {
+            return mockSession;
+        });
+    });
+
+    it("should redirect to /disruption-detail page after deleting disruptions for admin user", async () => {
```
[the remark this answers is not in the document]
Comment (alvak1) [reply site/pages/api/cancel-changes.test.ts]: done

File: site/pages/api/publish-edit.api.ts [hunk 2]
```diff
@@ -46,14 +50,40 @@ const publishEdit = async (req: NextApiRequest, res: NextApiResponse) => {
...
+
+        if (session.isOrgStaff && isEditPendingDsp)
+            await publishEditedConsequencesIntoPending(draftDisruption.disruptionId, session.orgId);
+
+        if (isEditPendingDsp && !session.isOrgStaff) {
+            await publishEditedConsequencesIntoPending(draftDisruption.disruptionId, session.orgId);
+            await publishPendingConsequences(draftDisruption.disruptionId, session.orgId);
+        }
+
+        if (!isEditPendingDsp) await publishEditedConsequences(draftDisruption.disruptionId, session.orgId);
```
[the remark this answers is not in the document]
Comment (alvak1) [reply site/pages/api/publish-edit.api.ts]: done

File: site/pages/api/reject.api.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+        if (!validatedBody.success || !session) {
+            redirectTo(res, ERROR_PATH);
+            return;
+        }
+
+        const draftDisruption = await getDisruptionById(validatedBody.data.disruptionId, session.orgId);
+
+        if (!draftDisruption || Object.keys(draftDisruption).length === 0) {
+            logger.error(`Disruption ${validatedBody.data.disruptionId} not found to publish`);
```
[the remark this answers is not in the document]
Comment (alvak1) [reply site/pages/api/reject.api.ts]: done

File: site/pages/api/reject.api.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+const reject = async (req: NextApiRequest, res: NextApiResponse) => {
+    try {
+        const validatedBody = publishSchema.safeParse(req.body);
+        const session = getSession(req);
+
+        if (!validatedBody.success || !session) {
+            redirectTo(res, ERROR_PATH);
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (alvak1) [reply site/pages/api/reject.api.ts:24]: done

File: site/constants/index.ts [hunk 5]
```diff
@@ -157,6 +157,18 @@ export const DISRUPTION_STATUSES: DisplayValuePair<Progress>[] = [
         value: Progress.open,
         display: "Open",
     },
+    {
+        value: Progress.pendingApproval,
+        display: "Pending Approval",
```
Comment (lj-kpmg) [inline site/constants/index.ts:162]:
```suggestion
        display: "Pending approval",
```

File: site/constants/index.ts [hunk 6]
```diff
@@ -157,6 +157,18 @@ export const DISRUPTION_STATUSES: DisplayValuePair<Progress>[] = [
...
+    },
+    {
+        value: Progress.editPendingApproval,
+        display: "Edit Pending Approval",
```
Comment (lj-kpmg) [inline site/constants/index.ts:166]:
```suggestion
        display: "Edit pending approval",
```

File: site/constants/index.ts (hunk 5 above)
Comment (alvak1) [reply site/constants/index.ts]: done

File: site/constants/index.ts (hunk 6 above)
Comment (alvak1) [reply site/constants/index.ts]: done

File: site/constants/index.ts [hunk 7]
```diff
@@ -157,6 +157,18 @@ export const DISRUPTION_STATUSES: DisplayValuePair<Progress>[] = [
...
+    },
+    {
+        value: Progress.draftPendingApproval,
+        display: "Draft Pending Approval",
```
[the remark this answers is not in the document]
Comment (alvak1) [reply site/constants/index.ts]: done

**Pull request merged**